### PR TITLE
Update styled-system types to 3.1 and update exports

### PR DESCRIPTION
**Pull request by antonvasin**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

All functions are exported from the [`index.js`](https://github.com/jxnblk/styled-system/blob/master/src/index.js) so it would be helpfull to be able to `import { px } from 'styled-system` instead of writing awkward sub-module import `import { px } from 'styled-system/dist/utils` or 
```
import { utils } from 'styled-system';
const { px } = utils;
```

Also `createMediaQuery` input type is the same as `num`’s, so I’ve updated that too.

### CODE REVIEW

File: types/styled-system/dist/util.d.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ export function merge(a: any, b: any): any;
 
 export function compose(...funcs: Array<(...args: any[]) => any>): (...args: any[]) => any;
 
-export function createMediaQuery(n: string): string;
+export function createMediaQuery(n: any): string;
```
Comment (alloy) [inline types/styled-system/dist/util.d.ts]: Can you type this as `number | string` instead?

**Review by alloy — CHANGES_REQUESTED** [review]
It actually looks like [v3.1 got rid of the separate modules](https://github.com/jxnblk/styled-system/pull/282), so these types should reflect that. Can you move all the declarations into the single index file, rather than re-exporting?

File: types/styled-system/dist/util.d.ts (hunk 1 above)
Comment (antonvasin) [reply types/styled-system/dist/util.d.ts]: Sure

**Comment by alloy** [comment]
Oops, missed the update! This looks good, thanks 👍

**Review by alloy — APPROVED** [review]

**Pull request merged**